### PR TITLE
feature/IISCRUM-2754

### PR DESCRIPTION
**Pull request by timyllym**
Check for empty RegExp /^$/ that is created by Grafana template variables and change it to /.*/ to enable getting all options

### CODE REVIEW

**Review by sapajune — COMMENTED** [review]
Change looks ok.

Since I'm not certain where regex part is actually injected (ticket image shows only var is specified) I'm worried this might trigger lots of large queries from panels during editing. Should it be restricted to cases where vars were actually replaced?

File: src/datasource-zabbix/datasource.ts [hunk 1]
```diff
@@ -731,10 +731,14 @@ export class ZabbixDatasource extends DataSourceApi<ZabbixMetricsQuery, ZabbixDS
     const parts = ['group', 'host', 'application', 'item'];
     _.forEach(parts, p => {
       if (target[p] && target[p].filter) {
+        const hasVars = this.checkForTemplateVariables(target[p].filter, options.scopedVars);
+        const origValue = target[p].filter;
         target[p].filter = this.replaceTemplateVars(target[p].filter, options.scopedVars);
```
Comment (sapajune) [inline src/datasource-zabbix/datasource.ts]: No need to call replacevars if no vars exist.

File: src/datasource-zabbix/datasource.ts [hunk 2]
```diff
@@ -754,6 +758,16 @@ export class ZabbixDatasource extends DataSourceApi<ZabbixMetricsQuery, ZabbixDS
     });
   }
 
+  checkForTemplateVariables(fieldText: string, scopedVars: any) {
+    let variablesFound = false;
+    Object.keys(scopedVars).map((variableName: string) => {
```
Comment (sapajune) [inline src/datasource-zabbix/datasource.ts]: This could be made more simple and shortcut capable by using some() instead.

**Review by sapajune — APPROVED** [review]
Check if replaceTemplateVariables() changed the target is probably unnecessary but code looks good now.

**Review by bulkware — APPROVED** [review]

**Pull request merged**